lib/python-indent.js

```diff
@@ -291,21 +294,33 @@ export default class PythonIndent {
         let indent = (this.editor.indentationForBufferRow(row)) +
             (atom.config.get("python-indent.hangingIndentTabs"));
 
-        // If the line where a newline was hit passes the DecreaseNextIndentPattern
-        // regex, then Atom will automatically decrease the indent on the next line
-        // but we don't actually want that since we're going to continue doing stuff
-        // on the next line.
-        const scope = this.editor.getRootScopeDescriptor();
-        const decreaseIndentPattern = this.editor.getDecreaseNextIndentPattern(scope);
-        const re = new RegExp(decreaseIndentPattern);
-
-        if (re.test(this.editor.lineTextForBufferRow(row - 1))) {
-            indent += 1;
-            if (row + 1 <= this.editor.buffer.getLastRow()) {
-                // If row+1 is a valid row, then that will also have been
-                // dedented, so we need to fix that too.
-                this.editor.setIndentationForBufferRow(row + 1,
-                    this.editor.indentationForBufferRow(row + 1) + 1);
+        // Use the old version of the "decreaseNextIndent" for Atom < 1.22
+        if (this.version[0] === 1 && this.version[1] < 22) {
+            // If the line where a newline was hit passes the DecreaseNextIndentPattern
+            // regex, then Atom will automatically decrease the indent on the next line
+            // but we don't actually want that since we're going to continue doing stuff
+            // on the next line.
+            const scope = this.editor.getRootScopeDescriptor();
+            const decreaseIndentPattern = this.editor.getDecreaseNextIndentPattern(scope);
+            const re = new RegExp(decreaseIndentPattern);
+
+            if (re.test(this.editor.lineTextForBufferRow(row - 1))) {
+                indent += 1;
+                if (row + 1 <= this.editor.buffer.getLastRow()) {
+                    // If row+1 is a valid row, then that will also have been
+                    // dedented, so we need to fix that too.
+                    this.editor.setIndentationForBufferRow(row + 1,
+                        this.editor.indentationForBufferRow(row + 1) + 1);
+                }
+            }
+        // Atom >= 1.22
+        } else {
+            const re = this.editor.tokenizedBuffer.decreaseNextIndentRegexForScopeDescriptor(
+                this.editor.getRootScopeDescriptor());
+
+            // If this is a "decrease next indent" line, then indent more
+            if (re.testSync(this.editor.lineTextForBufferRow(row - 1))) {
+                indent += 1;
             }
```

Comment:
Is the section with `if (row + 1 <= this.editor.buffer.getLastRow()) {` not needed any more?

I don't think I was actually able to get the spec to test for this behavior, but IIRC this block handled the last parentheses in:

```python
def f():
    return (
        x
    )
```

so that w/o the check for the next row you would get

```python
def f():
    return (
        x
)
```

Is that actually still happening? I'm not sure I can easily test from the beta version